src/daos-gen/daos-gen.c

```diff
@@ -0,0 +1,258 @@
+        }
+        rc = daos_cont_open(daos_args->src_poh, daos_args->src_cont_uuid,
+                            DAOS_COO_RW, &daos_args->src_coh, &co_info, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to open container: "DF_RC, DP_RC(rc));
+            rc = 1;
+        }
+    }
+
+    /* broadcast rank 0's pool and cont handle to everyone else */
+    daos_bcast_handle(rank, &(daos_args->src_poh), &(daos_args->src_poh), POOL_HANDLE); 
+    daos_bcast_handle(rank, &(daos_args->src_coh), &(daos_args->src_poh), CONT_HANDLE); 
+
+	daos_handle_t	    oh;
+	daos_array_iod_t    iod;
+	d_sg_list_t         sgl;
+	daos_range_t	    rg;
+	d_iov_t		        iov;
+	char		        buf[BUFLEN], rbuf[BUFLEN];
+	daos_size_t	        array_size;
+
+    /* TODO: generate objects/data inside of container */
+    rc = daos_obj_generate_oid(daos_args->src_coh, &oid, ofeats, OC_SX, 0, 0);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to generate oid: "DF_RC, DP_RC(rc));
+        rc = 1;
+    }
+
+	/** create the array */
+	rc = daos_array_create(daos_args->src_coh, oid, DAOS_TX_NONE, 1, 1048576, &oh,
+			               NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to create array: "DF_RC, DP_RC(rc));
+        rc = 1;
+    }
+
+	memset(buf, 'A', BUFLEN);
+
+	/** set array location */
+	iod.arr_nr = 1;
+	rg.rg_len = BUFLEN;
+	rg.rg_idx = 0;
+	iod.arr_rgs = &rg;
+
+	/** set memory location */
+	sgl.sg_nr = 1;
+	d_iov_set(&iov, buf, BUFLEN);
+	sgl.sg_iovs = &iov;
+
+	/** Write */
+	rc = daos_array_write(oh, DAOS_TX_NONE, &iod, &sgl, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to write to array: "DF_RC, DP_RC(rc));
+        rc = 1;
+    }
+
+	rc = daos_array_get_size(oh, DAOS_TX_NONE, &array_size, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to get array size: "DF_RC, DP_RC(rc));
+        rc = 1;
+    }
+
+    char uuid_str[130];
+    uuid_unparse(daos_args->src_cont_uuid, uuid_str);
+    printf("Successfully created container %s, array size: %d\n", uuid_str, (int)array_size);
+
+	rc = daos_array_close(oh, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to close array: "DF_RC, DP_RC(rc));
+        rc = 1;
+    }
+
+    /* don't close anything until all ranks are done using handles */
+    MPI_Barrier(MPI_COMM_WORLD);
+
+    tmp_rc = daos_cont_close(daos_args->src_coh, NULL);
+    MPI_Barrier(MPI_COMM_WORLD);
+    if (tmp_rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to close container (%d)", rc);
+        rc = 1;
+    }
+    
+    tmp_rc = daos_pool_disconnect(daos_args->src_poh, NULL);
+    MPI_Barrier(MPI_COMM_WORLD);
+    if (tmp_rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to disconnect pool (%d)", rc);
+        rc = 1;
+    }
+
+    /* Alert the user if there were copy errors */
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "One or more errors were detected while generating "
+                "data: " MFU_ERRF, MFU_ERRP(MFU_ERR_DAOS));
+    }
```

Comment:
@dsikich This is probably overkill here, since you're already checking error conditions above. The main reason I added this to some of the other tools is because errors aren't always logged and propagated through different layers of the MFU library